packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts

```diff
@@ -460,4 +462,100 @@ class DataStorageAuthWithTriggerTestProject extends TestProjectBase {
       throw err;
     }
   };
+
+  private createSQSQueue = async () => {
+    const createQueueResponse = await this.sqsClient.send(
+      new CreateQueueCommand({ QueueName: this.queueName })
+    );
+    this.queueUrl = createQueueResponse.QueueUrl ?? '';
+  };
+
+  private deleteSQSQueue = async () => {
+    if (this.queueUrl) {
+      await this.sqsClient.send(
+        new DeleteQueueCommand({
+          QueueUrl: this.queueUrl,
+        })
+      );
+    }
+  };
+
+  private addLambdaPermissionForSQS = async (backendId: BackendIdentifier) => {
+    this.funcWithSchedulePhysicalIds =
+      await this.resourceFinder.findByBackendIdentifier(
+        backendId,
+        'AWS::Lambda::Function',
+        (name) => name.includes('funcWithSchedule')
+      );
+    const getQueueAttributesResponse = await this.getSQSQueueAttributes(
+      this.queueUrl
+    );
+    const queueArn = getQueueAttributesResponse.Attributes?.QueueArn ?? '';
+
+    const sqsPolicy = new PolicyDocument({
+      statements: [
+        new PolicyStatement({
+          actions: ['sqs:SendMessage'],
+          effect: Effect.ALLOW,
+          resources: [queueArn],
+        }),
+      ],
+    });
+
+    const getFunctionConfigurationResponse = await this.lambdaClient.send(
+      new GetFunctionConfigurationCommand({
+        FunctionName: this.funcWithSchedulePhysicalIds[0],
+      })
+    );
+    const lambdaRole = getFunctionConfigurationResponse.Role
+      ? getFunctionConfigurationResponse.Role.split('/')[1]
+      : '';
+    await this.iamClient.send(
+      new PutRolePolicyCommand({
+        RoleName: lambdaRole,
+        PolicyName: 'LambdaSQSPolicy',
+        PolicyDocument: JSON.stringify(sqsPolicy),
+      })
+    );
+  };
+
+  private getSQSQueueAttributes = async (queueUrl: string) => {
+    const response = await this.sqsClient.send(
+      new GetQueueAttributesCommand({
+        QueueUrl: queueUrl,
+        AttributeNames: ['All'],
+      })
+    );
+
+    return response;
+  };
+
+  private assertScheduleInvokesFunction = async () => {
+    const TIMEOUT_MS = 1000 * 60 * 2; // 2 minutes
+    const startTime = Date.now();
+    let messageCount = 0;
+
+    // wait for schedule to invoke the function one time for it to send a message
+    while (Date.now() - startTime < TIMEOUT_MS && messageCount < 1) {
+      const response = await this.sqsClient.send(
+        new ReceiveMessageCommand({
+          QueueUrl: this.queueUrl,
+        })
+      );
+
+      if (response.Messages) {
+        messageCount += response.Messages.length;
+
+        // delete messages afterwards
+        for (const message of response.Messages) {
+          await this.sqsClient.send(
+            new DeleteMessageCommand({
+              QueueUrl: this.queueUrl,
+              ReceiptHandle: message.ReceiptHandle,
+            })
+          );
+        }
+      }
+    }
```

Comment:
Add a 10 sec wait in the loop, we wouldn't want to DoS the SQS service.

Reply:
OR use "long polling" , i.e. specify `WaitTimeSeconds` when calling receive message.

See https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-short-and-long-polling.html
and
https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/javascript_sqs_code_examples.html